### PR TITLE
Extract inner modules of `gpio` module into their own files

### PR DESCRIPTION
**Pull request by jessebraham**
Our GPIO driver is getting quite large and complex, and each of these modules have sufficient complexity to warrant their own files, IMO, so I have extracted the modules.

There is one breaking change here, in that I changed the `lp_gpio` module to `lp_io`. This was done for two reasons: to be consistent with the peripheral name (`LP_IO`) and to be consistent with the `rtc_io` module in the same driver. I can revert this change if needed.

### CODE REVIEW

File: esp-hal/src/gpio/lp_io.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        self.pulldown_enable(false);
+        self.into_pull_up_input().into_push_pull_output();
+
+        LowPowerPin {
+            private: PhantomData,
+        }
+    }
+}
+
+pub(crate) fn init_low_power_pin(pin: u8) {
```
[the remark this answers is not in the document]
Comment (jessebraham) [reply esp-hal/src/gpio/lp_io.rs:137]: `pub(crate)` items are not displayed in the documentation, so this one is fine to omit I think.

[the remark this answers is not in the document]
Comment (SergioGasquez) [reply esp-hal/src/gpio/lp_io.rs:137]: TIL, thanks <3

**Comment by jessebraham** [comment]
Good eye @SergioGasquez, thanks! Doc comment has been added.

I just realized there are a number of public items in the `gpio` module which still required doc comments as well; I am happy to take care of these too, but in the interest of getting this merged (also I have some other stuff on the go) I will probably break my new rule and do this in a separate PR 😬

**Review by bjoernQ — APPROVED** [review]
LGTM - I guess the amount of code in each module justifies having them separate

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

**Pull request merged**